sdk/eidolon_ai_sdk/apu/longterm_memory_unit.py

```diff
@@ -0,0 +1,143 @@
+from pydantic import BaseModel
+from typing import Optional, List
+from enum import Enum
+
+from eidolon_ai_sdk.agent_os import AgentOS
+from eidolon_ai_sdk.apu.call_context import CallContext
+from eidolon_ai_sdk.apu.llm_message import LLMMessage
+from eidolon_ai_sdk.system.reference_model import Specable
+from eidolon_ai_sdk.apu.processing_unit import ProcessingUnit
+from eidolon_ai_sdk.apu.llm_unit import LLMUnit
+from eidolon_ai_sdk.apu.mem0 import EidolonMem0
+from eidolon_ai_sdk.system.reference_model import Reference
+
+EIDOLON_DB_COL_NAME = "eidolon_mem0"
+
+
+class LongTermMemoryUnitScope(Enum):
+    SYSTEM = "system"
+    AGENT = "agent"
+    USER = "user"
+    USER_AGENT = "userAgent"
+    USER_PROCESS = "userProcess"
+
+
+class LongTermMemoryUnitConfig(BaseModel):
+    llm_unit: Optional[Reference[LLMUnit]]
+    pass
+
+
+class LongTermMemoryUnit(ProcessingUnit, Specable[LongTermMemoryUnitConfig]):
+    def __init__(self, default_llm: LLMUnit, unit_scope: LongTermMemoryUnitScope, spec: LongTermMemoryUnitConfig = None, **kwargs):
+        super().__init__(**kwargs)
+        self.spec = spec
+        self.unit_scope = unit_scope
+        if spec is not None and spec.llm_unit is not None:
+            self.mem0 = EidolonMem0(spec.llm_unit, EIDOLON_DB_COL_NAME)
+        else:
+            self.mem0 = EidolonMem0(default_llm, EIDOLON_DB_COL_NAME)
+
+    def storeMessage(self, call_context: CallContext, message: LLMMessage):
+        user_id = AgentOS.current_user().id
+        agent_name = AgentOS.current_agent_name()
+        metadata = {
+            "process_id": call_context.process_id,
+            "agent_name": agent_name
+        }
+        # not sure if I should manipulate result somehow - what's the expected return value?
+        return self.mem0.add(message, user_id=user_id, metadata=metadata)
```

Comment:
Should we be converting this to a string or does Luke's stuff handle it?

Reply:
i think mem0 itself handles it. i can convert to a string if that makes more sense, but it's worked this way so far